lib-extra/src/main/java/com/diffplug/spotless/extra/wtp/WtpEclipseFormatterStep.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.extra.wtp;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Properties;
+
+import com.diffplug.spotless.FormatterFunc;
+import com.diffplug.spotless.Provisioner;
+import com.diffplug.spotless.extra.EclipseBasedStepBuilder;
+
+/** Formatter step which calls out to the Groovy-Eclipse formatter. */
+public final class WtpEclipseFormatterStep {
+	// prevent direct instantiation
+	private WtpEclipseFormatterStep() {}
+
+	private static final String NAME = "eclipse wtp formatters";
+	private static final String FORMATTER_PACKAGE = "com.diffplug.spotless.extra.eclipse.wtp.";
+	private static final String DEFAULT_VERSION = "4.7.3a";
+	private static final String FORMATTER_METHOD = "format";
+
+	public static String defaultVersion() {
+		return DEFAULT_VERSION;
+	}
+
+	/** Provides default configuration for CSSformatter */
+	public static EclipseBasedStepBuilder createCssBuilder(Provisioner provisioner) {
+		return new EclipseBasedStepBuilder(NAME, " - css", provisioner, state -> apply("EclipseCssFormatterStepImpl", state));
+	}
```

Comment:
I might be missing something, but I think you can avoid adding the `formatterStepExt` variable to EclipseBasedStepBuilder and EclipseBasedStepBuilder.State if you apply the following change: 

```java
- new EclipseBasedStepBuilder(NAME, " - css", ...
+ new EclipseBasedStepBuilder(NAME + " - css", ...
```

Reply:
Currently I use `formatterName` to lookup the dependencies, which are equal for the different format methods (JSONS, JS, HTML...). So question remains whether I need a different name.
Each format method name per extension [must be unique](https://github.com/diffplug/spotless/blob/master/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/FormatExtension.java#L231). Actually I should have not made this change, but addressed the question instead, whether we want a distinct `FormatterExtension`.
For [typescript-formatter](https://github.com/diffplug/spotless/issues/119#issuecomment-413951442), there was the decision not to have a separate formatter extension.
But I understood that the `typescript-formatter` is one formatter for all. This is not the case for WTP. I can use the default WTP file extension settings to pick a formatter method for a file, but that should be configurable... So maybe it is worth to have dedicated `FormatterExtension`.

Reply:
> Currently I use formatterName to lookup the dependencies

Aha!  That's what I was missing.  In that case, this LGTM!

Reply:
Re: `FormatterExtension`, you means a `typescript {}` extension and a `javascript {}` extension (and `css {}` and `html {}`)?  I lean towards the default `FormatterExtension`, but I'm okay with having separate extensions instead.  Here is my argument:

"Convention-over-configuration" is a great idea, and gradle has strong conventions for all the JVM languages.  For the JVM languages we support (`java{}`, `kotlin{}`, `scala{}`), we can slurp the file locations from gradle's project model, and take advantage of that convention.  In my gradle web projects, I've found that there aren't strong conventions for the location of web resources.  Sometimes they're kinda scattered around in the `src/main/resources` directory, but other times they're being used for a static site outside of the JVM context, and they're just in the project root.  So we're not really able to use the "convention-over-configuration" to simplify setup for web resources.

On the other hand, it probably helps user-discoverability to have dedicated `html {}`, `css {}`, etc. blocks...  The real proof will be the docs.  If it's easier to document `html {}`, `css {}` - dedicated formats, then that's what we should do.

Reply:
OK, I'll remove the nae extension and aim for one `FormatterExtension` per WTP formatter method.